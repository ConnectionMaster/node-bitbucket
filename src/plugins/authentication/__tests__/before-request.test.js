const deepmerge = require('deepmerge')
const authenticate = require('../authenticate')
const beforeRequest = require('../before-request')

const state = {}

const initialEndpointOptions = {
  headers: {},
  url: '/test'
}

let endpointOptions

beforeEach(() => {
  state.auth = false
  endpointOptions = deepmerge({}, initialEndpointOptions)
})

describe('plugins:authentication/before-request', () => {
  it('does nothing if not authenticated', () => {
    beforeRequest(state, endpointOptions)
    expect(endpointOptions).toEqual(initialEndpointOptions)
  })

  describe.each([['basic'], ['app_password']])(
    'authentication `type`',
    type => {
      it('sets headers `authorization:Basic`', () => {
        let options = { type, username: 'username', password: 'password' }
        authenticate(state, options)
        beforeRequest(state, endpointOptions)
        expect(endpointOptions.headers.authorization).toMatchSnapshot()
      })
    }
  )

  describe.each([['token']])('authentication `type`', type => {
    it('sets headers `authorization:Bearer`', () => {
      let token = 'token'
      authenticate(state, { type, token })
      beforeRequest(state, endpointOptions)
      expect(endpointOptions.headers.authorization).toBe(`Bearer ${token}`)
    })
  })
})
