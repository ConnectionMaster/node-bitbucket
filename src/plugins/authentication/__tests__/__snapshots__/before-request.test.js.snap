// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins:authentication/before-request authentication \`type\` app_password sets headers \`authorization:Basic\` 1`] = `"Basic dXNlcm5hbWU6cGFzc3dvcmQ="`;

exports[`plugins:authentication/before-request authentication \`type\` basic sets headers \`authorization:Basic\` 1`] = `"Basic dXNlcm5hbWU6cGFzc3dvcmQ="`;
